### PR TITLE
Issue #15214: Migrated section 3.4.2.1 Overloads: never split to whole config testing

### PR DESCRIPTION
**Pull request by Zopsss**
#15214

### CODE REVIEW

File: src/it/resources-noncompilable/com/google/checkstyle/test/chapter3filestructure/rule3421overloadsplit/InputOverloadsNeverSplitRecords.java [hunk 1]
```diff
@@ -1,98 +1,101 @@
 //non-compiled with javac: Compilable with Java17
+
 package com.puppycrawl.tools.checkstyle.checks.coding.constructorsdeclarationgrouping;
 
+/** Testing. */
```
Comment (romani) [inline src/it/resources-noncompilable/com/google/checkstyle/test/chapter3filestructure/rule3421overloadsplit/InputOverloadsNeverSplitRecords.java:5]: `/** Some javadoc. */`

**Review by romani — CHANGES_REQUESTED** [review]
Apply to all

File: src/it/resources-noncompilable/com/google/checkstyle/test/chapter3filestructure/rule3421overloadsplit/InputOverloadsNeverSplitRecords.java (hunk 1 above)
Comment (Zopsss) [reply src/it/resources-noncompilable/com/google/checkstyle/test/chapter3filestructure/rule3421overloadsplit/InputOverloadsNeverSplitRecords.java]: done

**Comment by Zopsss** [comment]
failing CI: https://github.com/checkstyle/checkstyle/actions/runs/9862214638/job/27232334980?pr=15216

it is because the PR is referring to a closed issue, so no need to worry about it

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by Zopsss** [comment]
Strange why [Check no closed issue references / check_issues (pull_request)](https://github.com/checkstyle/checkstyle/actions/runs/9862214638/job/27239642089?pr=15216) is failing, PR is pointing to correct issue

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by romani** [comment]
Rebased by GitHub

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**